sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -232,6 +232,8 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper {
     case _ => JdbcUtils.getCommonJDBCType(dt)
   }
 
+  override def getDatabaseCalendar: Option[Calendar] = None
```

Comment:
Why MySQL dialect also return None here? the same as H2?

Reply:
Because MariaDB driver is used for MySQL and as far as I can see here [https://github.dev/mariadb-corporation/mariadb-connector-j/blob/75b1201bf57f2685bf[…]/internal/com/read/resultset/rowprotocol/BinaryRowProtocol.java](https://github.dev/mariadb-corporation/mariadb-connector-j/blob/75b1201bf57f2685bf49a11e7aa55f1e2e57f65d/src/main/java/org/mariadb/jdbc/internal/com/read/resultset/rowprotocol/BinaryRowProtocol.java#L1209) calendar is not considered (this corresponds to 2.7.9 that we use). 

Also, on their master they don't consider calendar too: [https://github.dev/mariadb-corporation/mariadb-connector-j/blob/0db5648c1b90303937[…]c/main/java/org/mariadb/jdbc/client/column/TimestampColumn.java](https://github.dev/mariadb-corporation/mariadb-connector-j/blob/0db5648c1b90303937b12929b592b8af96498bea/src/main/java/org/mariadb/jdbc/client/column/TimestampColumn.java#L99)

Reply:
Why not consider MySQL directly?

Reply:
Not sure what you mean. Like running it through mysql jdbc driver and not mariadb?